src/main/java/com/amplitude/Amplitude.java

```diff
@@ -108,11 +117,17 @@ public synchronized void flushEvents() {
       eventsToSend.clear();
       CompletableFuture.supplyAsync(
           () -> {
-            Response response = HttpCall.syncHttpCallWithEventsBuffer(eventsInTransit, apiKey);
+            Response response = null;
+            if (httpCallMode == Constants.httpCallMode.REGULAR_HTTPCALL) {
+              response =
+                  new GeneralHttpCall(eventsInTransit, apiKey).syncHttpCallWithEventsBuffer();
```

Comment:
I believe there is no need to initiate a new object for this each time...

Reply:
not a fan of put if else in here.. can we have a private method `getHttpCall`? Wondering if we should have a private variable to hold it.